src/webgpu/shader/execution/builtin/floor.spec.ts

```diff
@@ -0,0 +1,123 @@
+export const description = `
+Execution Tests for the 'floor' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, f32Bits, TypeF32 } from '../../../util/conversion.js';
+
+import { anyOf, kBit, kValue, run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('float_builtin_functions,floor')
+  .uniqueId('2edb4534aa0e48a6')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#float-builtin-functions')
+  .desc(
+    `
+floor:
+T is f32 or vecN<f32> floor(e: T ) -> T Returns the floor of e. Component-wise when T is a vector. (GLSLstd450Floor)
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    run(t, 'floor', [TypeF32], TypeF32, t.params, [
+      // Small positive numbers
+      { input: f32(0.1), expected: f32(0.0) },
+      { input: f32(0.9), expected: f32(0.0) },
+      { input: f32(1.0), expected: f32(1.0) },
+      { input: f32(1.1), expected: f32(1.0) },
+      { input: f32(1.9), expected: f32(1.0) },
+
+      // Small negative numbers
+      { input: f32(-0.1), expected: f32(-1.0) },
+      { input: f32(-0.9), expected: f32(-1.0) },
+      { input: f32(-1.0), expected: f32(-1.0) },
+      { input: f32(-1.1), expected: f32(-2.0) },
+      { input: f32(-1.9), expected: f32(-2.0) },
+
+      // Min and Max f32
+      { input: f32Bits(kBit.f32.negative.max), expected: f32(-1.0) },
+      { input: f32Bits(kBit.f32.negative.min), expected: f32Bits(kBit.f32.negative.min) },
+      { input: f32Bits(kBit.f32.positive.min), expected: f32(0.0) },
+      { input: f32Bits(kBit.f32.positive.max), expected: f32Bits(kBit.f32.positive.max) },
+
+      // Subnormal f32
+      { input: f32Bits(kBit.f32.subnormal.positive.max), expected: anyOf(f32(1.0), f32(0.0)) },
+      { input: f32Bits(kBit.f32.subnormal.positive.min), expected: anyOf(f32(1.0), f32(0.0)) },
+
+      // Infinity f32
+      { input: f32Bits(kBit.f32.infinity.negative), expected: f32Bits(kBit.f32.infinity.negative) },
```

Comment:
Infinities are now treated as UB in WGSL. We may want to remove these two tests.

Reply:
Done